pyvista/__init__.py

```diff
@@ -26,6 +26,12 @@
 
 from pyvista.utilities.wrappers import _wrappers
 
+# set up overrides once we're done with all the other imports
+from pyvista._overrides import set_up_overrides
+
+set_up_overrides()
+del set_up_overrides  # no longer necessary
```

Comment:
The current API is that we set up all subclasses of `DataSet` to be overrideable, and the only point of entry into this feature is `pv.PolyData.type_override` etc. class attributes. Hence my lack of ideas about documenting this feature beyond a dedicated docs page.